homeassistant/components/totalconnect/alarm_control_panel.py

```diff
@@ -172,84 +168,114 @@ def extra_state_attributes(self):
 
     async def async_alarm_disarm(self, code=None):
         """Send disarm command."""
-        await self.hass.async_add_executor_job(self._disarm)
-        await self.coordinator.async_request_refresh()
-
-    def _disarm(self, code=None):
-        """Disarm synchronous."""
         try:
-            ArmingHelper(self._partition).disarm()
+            await self.hass.async_add_executor_job(self._disarm)
+        except UsercodeInvalid as error:
+            self.coordinator.config_entry.async_start_reauth(self.hass)
+            raise HomeAssistantError(
+                "TotalConnect usercode is invalid. Did not disarm"
+            ) from error
         except BadResultCodeError as error:
             raise HomeAssistantError(
                 f"TotalConnect failed to disarm {self._name}."
             ) from error
+        await self.coordinator.async_request_refresh()
+
+    def _disarm(self, code=None):
+        """Disarm synchronous."""
+        ArmingHelper(self._partition).disarm()
 
     async def async_alarm_arm_home(self, code=None):
         """Send arm home command."""
-        await self.hass.async_add_executor_job(self._arm_home)
-        await self.coordinator.async_request_refresh()
-
-    def _arm_home(self):
-        """Arm home synchronous."""
         try:
-            ArmingHelper(self._partition).arm_stay()
+            await self.hass.async_add_executor_job(self._arm_home)
+        except UsercodeInvalid as error:
+            self.coordinator.config_entry.async_start_reauth(self.hass)
+            raise HomeAssistantError(
+                "TotalConnect usercode is invalid. Did not arm home"
+            ) from error
         except BadResultCodeError as error:
             raise HomeAssistantError(
                 f"TotalConnect failed to arm home {self._name}."
             ) from error
+        await self.coordinator.async_request_refresh()
+
+    def _arm_home(self):
+        """Arm home synchronous."""
+        ArmingHelper(self._partition).arm_stay()
```

Comment:
You could use functools.partial to avoid creating another function, but this is fine as well